### PR TITLE
feat: Support passing arguments to callback function in debounce & throttle function

### PR DESCRIPTION
**Pull request by jungwoo3490**
Hi :)
This PR resolves issue #18.

Previously, the debounce & throttle function could only handle callback functions without arguments.
With this update, the debounce & throttle function now properly handles callback function that require arguments.

## Changes
- Updated debounce & throttle function's type definition to accommodate callbacks that can receive arguments.
   ```ts
   // debounce.ts
   export function debounce<T extends (...args: any[]) => void>(func: T, debounceMs: number): { (...args: Parameters<T>): void; cancel: () => void } {
       ...
   }
   
   // throttle.ts
   export function throttle<T extends (...args: any[]) => void>(func: T, throttleMs: number) {
       ...
   }
   ```
- Adjusted the internal logic of debounce & throttle function to forward arguments to the callback function during invocation.
    
  ```ts
  // debounce.ts
  const debounced = function (...args: Parameters<T>) {
    if (timeoutId != null) {
      clearTimeout(timeoutId);
    }
    
    
    timeoutId = setTimeout(() => {
      func(...args);
    }, debounceMs);
  };
  
  // throttle.ts
  const throttledFunction = function (...args: Parameters<T>) {
    const now = Date.now();
        
    if (lastCallTime == null || now - lastCallTime >= throttleMs) {
      lastCallTime = now;
      func(...args);
    }
  }
  ```

### CODE REVIEW

File: src/function/debounce.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * have elapsed since the last time the debounced function was invoked. The debounced function also has a `cancel`
  * method to cancel any pending execution.
  *
- * @param {() => void} func - The function to debounce.
+ * @param {T} func - The function to debounce. T extends (...args: any[]) => void.
```
Comment (raon0211) [inline src/function/debounce.ts:6]:
```suggestion
 * @param {F extends (...args: any[]) => void} func - The function to debounce.
```

File: src/function/debounce.ts [hunk 2]
```diff
@@ -18,17 +18,17 @@
  * // Will not log anything as the previous call is canceled
  * debouncedFunction.cancel();
  */
-export function debounce(func: () => void, debounceMs: number): { (): void; cancel: () => void } {
+export function debounce<T extends (...args: any[]) => void>(func: T, debounceMs: number): { (...args: Parameters<T>): void; cancel: () => void } {
```
Comment (raon0211) [inline src/function/debounce.ts:21]:
```suggestion
export function debounce<F extends (...args: any[]) => void>(func: F, debounceMs: number): { (...args: Parameters<T>): void; cancel: () => void } {
```

File: src/function/debounce.ts [hunk 3]
```diff
@@ -18,17 +18,17 @@
...
   let timeoutId: number | NodeJS.Timeout | null = null;
 
-  const debounced = function () {
+  const debounced = function (...args: Parameters<T>) {
```
Comment (raon0211) [inline src/function/debounce.ts:24]:
```suggestion
  const debounced = function (...args: Parameters<F>) {
```

File: src/function/debounce.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@
  * have elapsed since the last time the debounced function was invoked. The debounced function also has a `cancel`
  * method to cancel any pending execution.
  *
- * @param {() => void} func - The function to debounce.
+ * @param {F extends (...args: any[]) => void} func - The function to debounce.
```
Comment (raon0211) [inline src/function/debounce.ts:6]:
```suggestion
 * @param {F} func - The function to debounce.
```

File: src/function/throttle.ts [hunk 5]
```diff
@@ -3,9 +3,9 @@
  * per every `throttleMs` milliseconds. Subsequent calls to the throttled function
  * within the wait time will not trigger the execution of the original function.
  *
- * @param {() => void} func - The function to throttle.
+ * @param {T} func - The function to throttle. T extends (...args: any[]) => void.
```
Comment (raon0211) [inline src/function/throttle.ts:6]:
```suggestion
 * @param {F} func - The function to throttle. T extends (...args: any[]) => void.
```

File: src/function/throttle.ts [hunk 6]
```diff
@@ -3,9 +3,9 @@
...
  * @param {number} throttleMs - The number of milliseconds to throttle executions to.
- * @returns {() => void} A new throttled function.
+ * @returns {(...args: Parameters<T>) => void} A new throttled function that accepts the same parameters as the original function.
```
Comment (raon0211) [inline src/function/throttle.ts:8]:
```suggestion
 * @returns {F} A new throttled function that accepts the same parameters as the original function.
```

File: src/function/debounce.ts [hunk 7]
```diff
@@ -3,7 +3,7 @@
...
  * @param {number} debounceMs - The number of milliseconds to delay.
  * @returns {{ (): void; cancel: () => void }} A new debounced function with a `cancel` method.
```
Comment (raon0211) [inline src/function/debounce.ts:8]:
```suggestion
 * @returns {F & { cancel: () => void }} A new debounced function with a `cancel` method.
```

File: src/function/throttle.ts [hunk 8]
```diff
@@ -23,15 +23,15 @@
  *   throttledFunction(); // Will log 'Function executed'
  * }, 1000);
  */
-export function throttle(func: () => void, throttleMs: number) {
+export function throttle<T extends (...args: any[]) => void>(func: T, throttleMs: number) {
```
Comment (raon0211) [inline src/function/throttle.ts:26]:
```suggestion
export function throttle<F extends (...args: any[]) => void>(func: F, throttleMs: number) {
```

File: src/function/throttle.ts [hunk 9]
```diff
@@ -23,15 +23,15 @@
...
   let lastCallTime: number | null;
 
-  const throttledFunction = function () {
+  const throttledFunction = function (...args: Parameters<T>) {
```
Comment (raon0211) [inline src/function/throttle.ts:29]:
```suggestion
  const throttledFunction = function (...args: Parameters<F>) {
```

File: src/function/debounce.ts [hunk 10]
```diff
@@ -18,17 +18,17 @@
...
     if (timeoutId != null) {
       clearTimeout(timeoutId);
     }
     
     
     timeoutId = setTimeout(() => {
-      func();
+      func(...args);
     }, debounceMs);
   };
```
Comment (raon0211) [inline src/function/debounce.ts:33]:
```suggestion
  } as F & { cancel: () => void };
```

**Comment by raon0211** [comment]
Thanks for your pull request! Meanwhile, I think we have to update our docs in our `docs` directory. We have the Korean version and the English version -- would you mind helping us?

File: src/function/throttle.ts [hunk 11]
```diff
@@ -23,15 +23,15 @@
...
-  const throttledFunction = function () {
+  const throttledFunction = function (...args: Parameters<F>) {
     const now = Date.now();
 
     if (lastCallTime == null || now - lastCallTime >= throttleMs) {
       lastCallTime = now;
-      func();
+      func(...args);
     }
   }
```
Comment (raon0211) [inline src/function/throttle.ts:36]:
```suggestion
  } as F;
```

**Comment by jungwoo3490** [comment]
> Thanks for your pull request! Meanwhile, I think we have to update our docs in our `docs` directory. We have the Korean version and the English version -- would you mind helping us?

I updated related docs 😀
Can you check them? @raon0211

**Review by raon0211 — APPROVED** [review]
Thanks, @jungwoo3490!

**Pull request merged**